conda_lock/src_parser/__init__.py

```diff
@@ -126,6 +207,10 @@ def __or__(self, other: "LockMeta") -> "LockMeta":
             channels=self.channels,
             platforms=sorted(set(self.platforms).union(other.platforms)),
             sources=ordered_union([self.sources, other.sources]),
+            time_metadata=other.time_metadata,
+            git_metadata=other.git_metadata,
+            inputs_metadata=other.inputs_metadata,
```

Comment:
This should combine the inputs_metadata dictionaries as they probably point to different files